streaming/stream_session.cc

```diff
@@ -193,20 +193,34 @@ void stream_manager::init_messaging_service_handler(abort_source& as) {
                     }
                 });
             };
+          using received_partitions_t = std::pair<uint64_t, std::unique_ptr<dht::auto_refreshing_sharder>>;
```

Comment:
I know why you are doing this but it is not nice. We can use a shared pointer and pass it around to take a ref instead.

Reply:
The object it keeps is created in a first lambda, I do not see how to achieve that with a shared_ptr